### ISSUE TITLE
Plot.Clear() doesn't clear everything

### ISSUE THREAD

**Issue by swharden** [swharden/ScottPlot#275]
This is a good example of a bad software pattern. Switch statements don't get updated when new pottables are added.

This hack works to clear experimental plottables (relates to #249):
```cs
wpfPlot1.plt.GetPlottables().Clear();
```

Perhaps Clear should only have two simple overloads:
* `Clear()` to clear all
* `Clear(Type)` to clear just that plottable type

**Comment by StendProg** [comment]
Switch statements always violates s**O**lid principe.

You can add optional predicate argument to `Clear()`.
This allow user make some specific selection, more flexible then only by type.
```c#
Clear(Predicate<IPlottable> selector = null)
{
    ...
    plottables.RemoveAll(selector)
}

and use it like:
Clear(x => x is PlottableBar || x is PlottableVSpan || x.Color == Color.Red)
```

**Comment by StendProg** [comment]
Switch statements always violates s**O**lid principe.

You can add optional predicate argument to `Clear()`.
This allow user make some specific selection, more flexible then only by type.
```c#
Clear(Predicate<IPlottable> selector = null)
{
    ...
    plottables.RemoveAll(selector)
}

and use it like:
Clear(x => x is PlottableBar || x is PlottableVSpan || x.Color == Color.Red)
```

**Comment by swharden** [comment]
> You can add optional predicate argument

I love this idea! Thanks for suggesting it. 

The new list of Clear() functions will be:

```cs
Plot.Clear() // to clear all
Plot.Clear(Plottable plottableType) // to clear just one type
Plot.Clear(Plottable[] plottableTypes) // to clear several types
Plot.Clear(Predicate<Plottable> selector) // to clear plots matching a criteria
```

**Comment by zrolfs** [comment]
I think this is happening for the new ErrorBarsPlottable that was added. Is there a quick fix?

**Comment by swharden** [comment]
> I think this is happening for the new ErrorBarsPlottable that was added. 

Thanks for bringing this to my attention! I'll prioritize fixing this and getting it on NuGet.

> Is there a quick fix?

Something like:
```cs
plt.GetPlottables().Clear()
```

(let me know if that does the trick)

**Comment by swharden** [comment]
> You can add optional predicate argument to `Clear()`.
> This allow user make some specific selection, more flexible then only by type.
> 
> ```cs
> Clear(Predicate<IPlottable> selector = null)
> {
>     ...
>     plottables.RemoveAll(selector)
> }
> 
> and use it like:
> Clear(x => x is PlottableBar || x is PlottableVSpan || x.Color == Color.Red)
> ```

@StendProg thanks for this tip! I added a `Clear()` method to do this. I also added some methods which accept `Type` and `Type[]`. I think this covers all use cases for `Clear()` now. What do you think?

https://github.com/swharden/ScottPlot/blob/e465cd311870c7d17dbc7b07411bef8bc7326b57/src/ScottPlot/Plot.cs#L234-L275

**Comment by swharden** [comment]
released in 4.0.24
